lib/rmt/cli/products.rb

```diff
@@ -35,21 +38,70 @@ def list
   end
   map ls: :list
 
-  desc 'enable', 'Enable mirroring of product repositories by product ID or product string.'
+  desc 'enable TARGETS', 'Enable mirroring of product repositories by a list of product IDs or product strings.'
   option :all_modules, type: :boolean, desc: 'Enables all free modules for a product'
-  def enable(target)
-    change_product(target, true, options[:all_modules])
+  def enable(*targets)
+    change_products(targets, true, options[:all_modules])
   end
 
-  desc 'disable', 'Disable mirroring of product repositories by product ID or product string.'
-  def disable(target)
-    change_product(target, false, false)
+  desc 'disable TARGETS', 'Disable mirroring of product repositories by a list of product IDs or product strings.'
+  def disable(*targets)
+    change_products(targets, false, false)
   end
 
   protected
 
+  def change_products(targets, set_enabled, all_modules)
+    success = true
+    targets = clean_target_input(targets)
+    raise RMT::CLI::Error.new('No product ids supplied') if targets.empty?
+
+    targets.each do |target|
+      change_product(target, set_enabled, all_modules)
+    rescue ProductNotFoundException => e
+      puts e.message
+      success = false
+    end
+
+    raise RMT::CLI::Error.new("Not all products were #{set_enabled ? 'enabled' : 'disabled'}.") unless success
```

Comment:
I think a message at the end saying "Not all products were enabled" with no specifics about /which/ products will lead to confusion and doubt.

I would go for something like "Products X, Y, and Z were not enabled, because they were not found."